client/src/client.rs

```diff
@@ -61,16 +63,36 @@ impl Client {
 
     /// Register a new account with the server.
     /// `phone_number` must be unique.
-    pub async fn register(name: &str, phone_number: String) -> Result<Self, SignalClientError> {
+    pub async fn register(
+        name: &str,
+        phone_number: String,
+    ) -> Result<Client<Device>, SignalClientError> {
         let mut csprng = OsRng;
         let aci_registration_id = OsRng.gen_range(1..16383);
         let pni_registration_id = OsRng.gen_range(1..16383);
         let aci_key_pair = KeyPair::generate(&mut csprng);
         let pni_key_pair = KeyPair::generate(&mut csprng);
         let id_key = IdentityKey::new(aci_key_pair.public_key);
         let id_key_pair = IdentityKeyPair::new(id_key, aci_key_pair.private_key);
+        dotenv().map_err(|err| SignalClientError::DatabaseError(format!("{err}")))?;
```

Comment:
This isnt nessesarily a database error, no?

Reply:
Well, technically no